src/dlgTriggerEditor.cpp

```diff
@@ -6769,6 +6991,89 @@ void dlgTriggerEditor::slot_export()
     };
 }
 
+void dlgTriggerEditor::slot_copy_xml()
+{
+    switch( mCurrentView )
+    {
+    case cmTriggerView:
+        exportTriggerToClipboard( );
+        break;
+    case cmTimerView:
+        exportTimerToClipboard( );
+        break;
+    case cmAliasView:
+        exportAliasToClipboard( );
+        break;
+    case cmScriptView:
+        exportScriptToClipboard( );
+        break;
+    case cmActionView:
+        exportActionToClipboard( );
+        break;
+    case cmKeysView:
+        exportKeyToClipboard( );
+        break;
+    };
+}
+
+void dlgTriggerEditor::slot_paste_xml()
+{
+    if( mCurrentView )
+    {
+        switch( mCurrentView )
+        {
+            case cmTriggerView:
+                saveTrigger();
+                break;
+            case cmTimerView:
+                saveTimer();
+                break;
+            case cmAliasView:
+                saveAlias();
+                break;
+            case cmScriptView:
+                saveScript();
+                break;
+            case cmActionView:
+                saveAction();
+                break;
+            case cmKeysView:
+                saveKey();
+                break;
+        };
+    }
+    QString profileName = mpHost->getName();
+    QString login = mpHost->getLogin();
+    QString pass = mpHost->getPass();
```

Comment:
These three lines and their counter-parts after `reader.importFromClipBoard()` are likely to be redundant as they are only to prevent those data items being overwritten by an XMLimport of an entire profile where the `<Host ...>` and its sub-elements can have new/different values for them. It depends on just what could be picked up as a source of the XML data on the clipboard. {e.g. a Post-mortem examination of a profile's saved data file opened up in a  text editor or some Frankensteinian experimentation with such things :scream:}